### ISSUE TITLE
nnet3-info occurs a segment fault.

### ISSUE THREAD

**Issue by hwiorn** [kaldi-asr/kaldi#2875]
Using kaldi latest (70aba21bf1a3cd23a2bb69bf37a22f2505a6e126), I get the following error when running nnet3-info. (It happens during steps/nnet3/xconfig_to_configs.py)

```
$ nnet3-info exp/chain_rvb/tdnn1n_rvb/configs//ref.raw
Segmentation fault (core dumped)
```

src/nnet3/nnet-parse.cc:522 <- Infinite recursion occurs at this location.

The error is a side effect caused by #2496 and is associated with a commit of c362b2cf4bfbe51282645a472f04a76e8ee475f5.
The following changes are required.

```
diff --git a/src/nnet3/nnet-parse.cc b/src/nnet3/nnet-parse.cc
index c920790..a51bba2 100644
--- a/src/nnet3/nnet-parse.cc
+++ b/src/nnet3/nnet-parse.cc
@@ -518,12 +518,12 @@ std::string SummarizeVector(const VectorBase<float> &vec) {
 }

 std::string SummarizeVector(const VectorBase<double> &vec) {
-  Vector<BaseFloat> vec_copy(vec);
+  Vector<float> vec_copy(vec);
   return SummarizeVector(vec_copy);
 }

 std::string SummarizeVector(const CuVectorBase<BaseFloat> &cu_vec) {
-  Vector<BaseFloat> vec(cu_vec);
+  Vector<float> vec(cu_vec);
   return SummarizeVector(vec);
 }
```

**Comment by danpovey** [comment]
@huangruizhe  can you please make a PR for a fix?

**Comment by huangruizhe** [comment]
Fixed in PR #2876
